judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/submission/programming/ContestSubmissionResource.java

```diff
@@ -228,6 +234,29 @@ public SubmissionWithSourceResponse getSubmissionWithSourceById(
                 .build();
     }
 
+    @Override
+    @UnitOfWork(readOnly = true)
+    public Response getSubmissionSourceImage(String contestJid, String userJid, String problemJid) {
+        Contest contest = checkFound(contestStore.getContestByJid(contestJid));
+        checkAllowed(scoreboardRoleChecker.canViewOtherContestantSolution(contest));
+
+        List<Submission> submissions = submissionStore
+                .getSubmissions(Optional.of(contestJid), Optional.of(userJid), Optional.of(problemJid), Optional.of(1))
+                .getPage();
+        if (submissions.isEmpty()) {
+            return Response.noContent().build();
+        }
+
+        Submission submission = submissions.get(0);
+        String username = this.userClient.getProfile(userJid).getUsername();
+        SubmissionSource source = submissionSourceBuilder.fromPastSubmission(submission.getJid());
+
+        String rawSource = new String(source.getSubmissionFiles().get(SubmissionSource.DEFAULT_KEY).getContent());
```

Comment:
This won't work if the problem has multiple source keys (e.g. IOI problems with 2 submission files) :')

Can we iterate the map instead, to show all sources? Similar to how we show submission in submission details page.